Mathlib/Analysis/Complex/UpperHalfPlane/Basic.lean

```diff
@@ -29,10 +29,10 @@ we state lemmas in this file without spurious `coe_fn` terms. -/
 attribute [-instance] Matrix.SpecialLinearGroup.instCoeFun
 attribute [-instance] Matrix.GeneralLinearGroup.instCoeFun
 
-local notation "GL(" n ", " R ")" "⁺" => Matrix.GLPos (Fin n) R
-local notation:1024 "↑ₘ" A:1024 =>
+local notation:1024 "↑ₘₙ" A:1024 =>
```

Comment:
I still find this very confusing. We seem to have _local_ notations `↑ₘₙ` and `↑ₘₙ[R]` defined here, and also _scoped_ (non-local) notations with the same symbols in `SlashActions.lean` but i can't quite tell if they're actually the same; and we now have `↑ₘ` (without `n`) as well, defined in yet another file.

Can you add docstrings to the ones which don't currently have them, and cross-reference between the docstrings to explain how they are different?

Reply:
Ah yes I should add the doc string here also. It seems we had `↑ₘ` meaning at least 2 different things, one was coercing through SL to matrices and the other was via GL_pos to GL, to matrices. I left the former as ↑ₘ and the latter is now ↑ₘₙ (the mn for many coercions I guess as it wouldnt give me a small g). I made them scoped in the ModularForms and ModularGroup namespaces.